docs/proposals/user-namespace.md

```diff
@@ -0,0 +1,226 @@
+<!-- BEGIN MUNGE: UNVERSIONED_WARNING -->
+
+<!-- BEGIN STRIP_FOR_RELEASE -->
+
+<img src="http://kubernetes.io/kubernetes/img/warning.png" alt="WARNING"
+     width="25" height="25">
+<img src="http://kubernetes.io/kubernetes/img/warning.png" alt="WARNING"
+     width="25" height="25">
+<img src="http://kubernetes.io/kubernetes/img/warning.png" alt="WARNING"
+     width="25" height="25">
+<img src="http://kubernetes.io/kubernetes/img/warning.png" alt="WARNING"
+     width="25" height="25">
+<img src="http://kubernetes.io/kubernetes/img/warning.png" alt="WARNING"
+     width="25" height="25">
+
+<h2>PLEASE NOTE: This document applies to the HEAD of the source tree</h2>
+
+If you are using a released version of Kubernetes, you should
+refer to the docs that go with that version.
+
+Documentation for other releases can be found at
+[releases.k8s.io](http://releases.k8s.io).
+</strong>
+--
+
+<!-- END STRIP_FOR_RELEASE -->
+
+<!-- END MUNGE: UNVERSIONED_WARNING -->
+
+# User Namespaces
+
+## Motivation
+
+Running a container as root is a common practice for many
+image authors. However, many clusters operate under a more strict
+security profile that prevents usage of the root UID. This means that
+either admins must allow exceptions for root access to run many of
+the images available or image authors must be aware of the deployment
+environment in order to develop a container that works well under
+the increased security constraints. This frustrates end-users that
+want to run the same images they run on their host as they cannot
+run those images on the cluster.
+
+Docker 1.10 supports [user namespace remapping](https://docs.docker.com/v1.10/engine/reference/commandline/daemon/#daemon-user-namespace-options)
+in the docker daemon.  This allows you to run as the root user in a
+container while running as a non-root user in the host.
+
+Additionally, the `HostConfig` for a container now contains a parameter
+that lets you escape the private user namespace and indicate that you
+would like to [use the host's user namespace](https://github.com/docker/engine-api/blob/master/types/container/host_config.go#L310).
```

Comment:
I suggest to use URL to a certain commit and protect from the future changes. For example: https://github.com/docker/engine-api/blob/4290f40c056686fcaa5c9caf02eac1dde9315adf/types/container/host_config.go#L310